### PR TITLE
Use String as id for Storable

### LINKED ISSUE #26: Support id as string
should we?

### PR DESCRIPTION
**Pull request by MP0w**
Closes #26 

Rationale: 
mpow [5:24 PM]  
FYI just implemented id as String instead of Int otherwise JSONAPIEntity can't be easily Storable, the id generated by db will still be Int converted to String. This is better because Int is always convertible to string but not the other way around

 (docu missing, waiting for merge of docu PR)

### CODE REVIEW

**Comment by MP0w** [comment]
Will merge it for the moment as I need the changes, we cans till discuss later and eventually revert if you don't agree.

File: Source/KakapoDB.swift [hunk 1]
```diff
@@ -14,9 +14,18 @@ import Foundation
  This is a base protocol and it's only used internally, for your objects you should check `Storable` instead.
  */
 public protocol _Storable {
-    // TODO: default to String and create a protocol that as an Int id with default implementation. Because String is not always convertible to Int but Int can always be converted to String.
-    var id: Int { get }
-    init(id: Int, db: KakapoDB)
+    /// The unique identifier provided by `KakapoDB`, objects shouldn't generate ids themselves. `KakapoDB` generate `Int` ids converted to String for better compatibilities with standards like JSONAPI, in case you need `Int` ids is safe to ssume that the conversion will always succeeed.
```
Comment (joanromano) [inline Source/KakapoDB.swift:17]: _assume_

Comment (joanromano) [inline Source/KakapoDB.swift:17]: I think we could provide an example with Int identifiers here.

File: Source/KakapoDB.swift [hunk 2]
```diff
@@ -14,9 +14,18 @@ import Foundation
...
+    var id: String { get }
+    
+    /**
+     An initializer that is used by `KakapoDB` to create objects to be stored in the db
+     
+     - parameter id: The unique identifier provided by `KakapoDB`, objects shouldn't generate ids themselves. `KakapoDB` generate `Int` ids converted to String for better compatibilities with standards like JSONAPI, in case you need `Int` ids is safe to ssume that the conversion will always succeeed.
+     - parameter db: The db that is creating the object, can be used  to generate other `Storable` objects, for example relationsips of the object by using `db.create(MyRelationshipType)`.
```
Comment (joanromano) [inline Source/KakapoDB.swift:24]: Not so clear why would you need the db param for

**Comment by joanromano** [comment]
👍

**Pull request merged**